### PR TITLE
Add remane cookie variable

### PR DESCRIPTION
**Pull request by xorgxx**
To be possible to modify "Cookie_Consent" and "Cookie_Category" by a variable in the config.yml
  #27 
```
ch_cookie_consent:
    theme: 'dark' # light, dark
    categories: # Below are the default supported categories
        - 'analytics'
        - 'tracking'
        - 'marketing'
        - 'social_media'
    use_logger: false # Logs user actions to database

    # adding this in the config file <----------------------------
    cookie_prefix: 'neo_Consent'                                          # 'cookie_Consent'
    cookie_key_name: 'neo_Consent_Key'                                    # 'cookie_Consent_Key'
    cookie_consent_name: 'neo_Category_'                                  # 'cookie_Category_'
```

### CODE REVIEW

**Comment by matthijsch** [comment]
Isn't it sufficient to be able to change the prefix of the cookie names? In that case you only have to add the cookie_prefix option and that would leave the code much cleaner. Right now the cookiePrefix isn't used anywhere. Also I think only in the Enum class the custom prefix should be injected

**Comment by xorgxx** [comment]
Yes cookie_prefix  did not see where to use, in first analyse. As you, maybe it's better to use only cookie_prefix and inject in Enum class but are making injection  ? we cant modifie const, i see only one way : to create service CookieNameEnum with gutters and setters and inject this service where it needs. But this solution is almost the identical to me.

**Comment by xorgxx** [comment]
```
class CookieNameEnum
{
    const COOKIE_CONSENT_NAME           = 'cookie_Consent';
    const COOKIE_CONSENT_KEY_NAME       = 'cookie_Consent_Key';
    const COOKIE_CATEGORY_NAME_PREFIX   = 'cookie_Category_';

    private $cookiePrefix;

    /**
     * @var array
     */
    public static $prefixCookie = [
        "COOKIE_CONSENT_NAME"           => "",
        "COOKIE_CONSENT_KEY_NAME"       => "",
        "COOKIE_CATEGORY_NAME_PREFIX"   => "",
    ];

    private static $Prefix;

    public function __construct(string $cookiePrefix) 
    {
        self::$Prefix       = $cookiePrefix;
        self::$prefixCookie = [
            "COOKIE_CONSENT_NAME"           => self::$Prefix . self::COOKIE_CONSENT_NAME ,
            "COOKIE_CONSENT_KEY_NAME"       => self::$Prefix . self::COOKIE_CONSENT_KEY_NAME ,
            "COOKIE_CATEGORY_NAME_PREFIX"   => self::$Prefix . self::COOKIE_CONSENT_KEY_NAME ,
            ];
    }
    /**
     * Get all cookie consent prefix (this was only for test).
     */
    public static function getAvailableCookie ( ){

        self::$prefixCookie = [
            "COOKIE_CONSENT_NAME"           => self::$Prefix . self::COOKIE_CONSENT_NAME ,
            "COOKIE_CONSENT_KEY_NAME"       => self::$Prefix . self::COOKIE_CONSENT_KEY_NAME ,
            "COOKIE_CATEGORY_NAME_PREFIX"   => self::$Prefix . self::COOKIE_CONSENT_KEY_NAME ,
        ];

        return self::$prefixCookie;
    }

    /**
     * Get cookie category name.
     */
    public static function getCookieCategoryName(string $category): string
    {
        return self::$Prefix . self::COOKIE_CATEGORY_NAME_PREFIX.$category;
    }
}
```
and for exemple CookieChecker : 

```
    /**
     * CookieChecker constructor.
     *
     * @param Request $request

     */
    public function __construct(Request $request, CookieNameEnum $cookieNameEnum) 
    {
        $this->request              = $request;
        $this->cookiePrefix         = $cookieNameEnum;
    }
.....
    /**
     * @return bool
     */
    public function isCookieConsentSavedByUser(  ): bool
    {
        return $this->request->cookies->has( CookieNameEnum::$prefixCookie["COOKIE_CONSENT_NAME"]);
    }
```

File: Cookie/CookieChecker.php [hunk 1]
```diff
@@ -19,25 +19,55 @@ class CookieChecker
...
+        $this->cookiePrefix         = $cookiePrefix;
+        $this->cookieKeyName        = $cookieKeyName;
+        $this->cookieConsentName    = $cookieConsentName;
     }
 
     /**
      * Check if cookie consent has already been saved.
      */
-    public function isCookieConsentSavedByUser(): bool
+//    public function isCookieConsentSavedByUser(  ): bool
```
Comment (reyostallenberg) [inline Cookie/CookieChecker.php]: This commented code should be removed

File: Enum/CookieNameEnum.php [hunk 2]
```diff
@@ -11,17 +11,17 @@
 
 class CookieNameEnum
 {
-    const COOKIE_CONSENT_NAME = 'Cookie_Consent';
+    const COOKIE_CONSENT_NAME = 'cookie_Consent';
```
Comment (reyostallenberg) [inline Enum/CookieNameEnum.php]: This is a none BC change

File: README.md [hunk 3]
```diff
@@ -49,6 +49,11 @@ ch_cookie_consent:
         - 'marketing'
         - 'social_media'
     use_logger: true # Logs user actions to database
+    
+    # add trying
```
Comment (reyostallenberg) [inline README.md]: There is no explanation whatsoever about why 'neo' is here, you should explain what these settings do, why they are possible/needed and what they default to.

File: Resources/config/services.yaml [hunk 4]
```diff
@@ -18,8 +21,13 @@ services:
         resource: '../../Controller/*'
         public: true
 
+    ConnectHolland\CookieConsentBundle\EventSubscriber\:
+        resource: '../../EventSubscriber/*'
+        public: true
```
Comment (reyostallenberg) [inline Resources/config/services.yaml]: Please explain this change

File: Resources/config/services.yaml [hunk 5]
```diff
@@ -18,8 +21,13 @@ services:
...
+
     ConnectHolland\CookieConsentBundle\Cookie\CookieChecker:
-        arguments: ["@=service('request_stack').getCurrentRequest()"]
+        arguments:
```
Comment (reyostallenberg) [inline Resources/config/services.yaml]: Please explain this style change

**Comment by reyostallenberg** [comment]
Also:

- Please run `./bin/php-cs-fixer fix` before committing your contribution
- Make the checks green

File: DependencyInjection/Configuration.php [hunk 6]
```diff
@@ -18,14 +18,8 @@ class Configuration implements ConfigurationInterface
 {
     public function getConfigTreeBuilder(): TreeBuilder
     {
-        $treeBuilder = new TreeBuilder('ch_cookie_consent');
-        
-        if (method_exists($treeBuilder, 'getRootNode')) {
-            $rootNode = $treeBuilder->getRootNode();
-        } else {
-            // BC layer for symfony/config 4.1 and older
```
Comment (reyostallenberg) [inline DependencyInjection/Configuration.php]: Do not remove this BC layer

File: Resources/config/services.yaml (hunk 4 above)
Comment (xorgxx) [reply Resources/config/services.yaml]: This in last version is remove.

File: Resources/config/services.yaml (hunk 5 above)
Comment (xorgxx) [reply Resources/config/services.yaml]: This in last version is remove.

File: README.md (hunk 3 above)
Comment (xorgxx) [reply README.md]: "neo" was for my internal testing.

**Comment by xorgxx** [comment]
in the last version i have include : 
```
class CookieNameEnum
{
    const COOKIE_PREFIX                 = 'cookie_';
    const COOKIE_CONSENT_NAME           = 'Consent';
    const COOKIE_CONSENT_KEY_NAME       = '_Key';
    const COOKIE_CATEGORY_NAME_PREFIX   = 'Category_';

    private $cookiePrefix;

    /**
     * @var array
     */
    public static $prefixCookie = [
        "COOKIE_CONSENT_NAME"           => "",
        "COOKIE_CONSENT_KEY_NAME"       => "",
        "COOKIE_CATEGORY_NAME_PREFIX"   => "",
    ];

    private static $Prefix;
    private static $Consent_name;
    private static $Consent_key;
    private static $Consent_category;

    public function __construct(string $cookiePrefix, string $cookieKeyName, string $cookieConsentName, string $cookieConsentCategory) //, string $cookiePrefix
    {
        self::$Prefix               = $cookiePrefix;
        self::$Consent_name         = $cookieConsentName;
        self::$Consent_key          = $cookieKeyName;
        self::$Consent_category     = $cookieConsentCategory;

        self::$prefixCookie = [
            "COOKIE_CONSENT_NAME"           => self::$Prefix . self::$Consent_name ,
            "COOKIE_CONSENT_KEY_NAME"       => self::$Prefix . self::$Consent_name . self::$Consent_key ,
            "COOKIE_CATEGORY_NAME_PREFIX"   => self::$Prefix . self::$Consent_category ,
            ];
    }
    /**
     * Get all cookie consent prefix (this was only for test).
     */
    public static function getAvailableCookie ( ){

        self::$prefixCookie = [
            "COOKIE_CONSENT_NAME"           => self::$Prefix . self::$Consent_name ,
            "COOKIE_CONSENT_KEY_NAME"       => self::$Prefix . self::$Consent_name . self::$Consent_key ,
            "COOKIE_CATEGORY_NAME_PREFIX"   => self::$Prefix . self::$Consent_category ,
        ];

        return self::$prefixCookie;
    }

    /**
     * Get cookie category name.
     */
    public static function getCookieCategoryName(string $category): string
    {
        return self::$Prefix . self::$Consent_category.$category;
    }
}
```
i think that it can be gainful and it's include. it das not take must.

**Pull request closed without merging**